### DISCUSSION TITLE
Using Flight Tracker with FBW a320:   accessing FBW SimConnect variables

### DISCUSSION THREAD

**Discussion by Rascal650** [nguyenquyhy/Flight-Tracker-StreamDeck#381] [General]
Hi, I started using Flight Deck to program some cockpit panels on the FBW a320.  

Right now, I'm programming the different ECAM pages (ENG, PRESS, BLEED, etc) in the central pedestal.  I found the right MOBIFLIGHT variables (ex:  MOBIFLIGHT_A320_Neo_EICAS_2_ECAM_CHANGE_PAGE_BLEED to click on the BLEED button) and I can successfully change the ECAM pages from the stream deck.

I would like to change the background icons when the button is selected (a LED shows a white rectangle) vs. when the button is not selected (the LED for the button is off; NO white rectangle).  See attached snapshot:
![ecam-snapshot](https://user-images.githubusercontent.com/88073848/210009300-8b1daa83-0d58-4e04-ace7-a8c786978c48.jpg)

Using the MSFS DEV mode, I found this variable  A32NX_ECAM_SD_CURRENT_PAGE_INDEX  which is set to 0 for ENG, 1 for BLEED, 2 for PRESS and so forth.

I cannot seem to use this variable in the "Feedback Value" field.   I tried A32NX_ECAM_SD_CURRENT_PAGE_INDEX == 1  for the BLEED button.   That did not work; my background icons do not change.

Is that supported?  Thanks! for the support.

**Comment by nguyenquyhy** [comment]
My guess is that A32NX_ECAM_SD_CURRENT_PAGE_INDEX is an L-Var, which is unfortunately not supported in the main plugin yet. There is already an issue tracking this feature https://github.com/nguyenquyhy/Flight-Tracker-StreamDeck/issues/218 but I cannot manage time to get to it.

In brief, L-Vars are not sent to external plugins via SimConnect, so the plugin cannot outright read that value. Instead, it will need an internal WASM mod to read the values to send them out. The direction now is to leverage that capability of Mobiflight too.

There is already a fork that implement the feature (https://github.com/nguyenquyhy/Flight-Tracker-StreamDeck/pull/286), but I want to implement it a bit differently so I have not merged that change for now.

**Reply by Rascal650** [reply]
Thanks for the reply.  I checkd the FBW documentation and indeed, A32NX_ECAM_SD_CURRENT_PAGE_INDEX  is a L-Var.

**Comment by Rascal650** [comment]
I noticed that Mobiflight provides lots of output variables (coded and updated via the HubHop database).  

Once of them is A32NX_ECAM_ACTIVE_PAGE_DOOR that returns TRUE if the DOOR button is selected on the ECAM page. You mentioned that LVar are not accessible from Flight Tracker but would this Mobiflight A32NX_ECAM_ACTIVE_PAGE_DOOR be accessible in the Feedback Value field?    If so, what is the syntax?  I tried MOBILFIGHT_A32NX_ECAM_ACTIVE_PAGE_DOOR  but that's not working.

Thanks!

**Reply by rmroc451** [reply]
Looks like in the variable you posted that you used, mobiflight is misspelled. That should work, if spelled correctly! Happy flying!

`MOBIFLIGHT_A32NX_ECAM_ACTIVE_PAGE_DOOR`

**Reply by Rascal650** [reply]
Thanks for the reply. You are right; I misspelled it in this discussion thread but it's correctly spelled in Flight Tracker. I tried to to add these variables (one for ENG, BLEED, PRESS, ELEC and so on) in all the Feedback value fields but my background images do not change.  These variables are visible from Mobiflight [as SimConnect(MSFS2020)] as "Output".   Does Flight Tracker gets access to these Mobiflight variables?  If so, what is the syntax I need to use in the Feedback value field.  FYI: These variables are boolean.  Thanks for the help!
